proxy/src/coco.rs

```diff
@@ -1,649 +1,291 @@
-    };
-
-    let mut branches = browser
-        .list_branches(None)
-        .expect("Getting branches failed")
-        .into_iter()
-        .map(|b| Branch(b.name.name().to_string()))
-        .collect::<Vec<Branch>>();
-
-    branches.sort();
-
-    Ok(branches)
-}
-
-/// Given a path to a repo returns the list of branches.
-///
-/// # Errors
-///
-/// Will return [`error::Error`] if the repository doesn't exist.
-pub fn local_branches(repo_path: &str) -> Result<Vec<Branch>, error::Error> {
-    let repo = surf::git::Repository::new(repo_path)?;
-    let browser = surf::git::Browser::new(repo)?;
-    let mut branches = browser
-        .list_branches(None)?
-        .into_iter()
-        .map(|b| Branch(b.name.name().to_string()))
-        .collect::<Vec<Branch>>();
-
-    branches.sort();
-
-    Ok(branches)
-}
-
-/// Retrieves the [`Commit`] for the given `sha1`.
-///
-/// # Errors
-///
-/// Will return [`error::Error`] if the project doesn't exist or the surf interaction fails.
-pub fn commit(paths: &Paths, id: &str, sha1: &str) -> Result<Commit, error::Error> {
-    let project_id = project::ProjectId::from_str(id)?;
-    let project = project::Project::open(paths, &project_id)?;
-    let mut browser = match project {
-        project::Project::Git(git_project) => git_project.browser()?,
-    };
-
-    browser.commit(surf::git::Oid::from_str(sha1)?)?;
-
-    let history = browser.get();
-    let commit = history.first();
 
-    Ok(Commit::from(commit))
-}
-
-/// Retrieves the [`Commit`] history for the given `branch`.
-///
-/// # Errors
-///
-/// Will return [`error::Error`] if the project doesn't exist or the surf interaction fails.
-pub fn commits(paths: &Paths, id: &str, branch: &str) -> Result<Vec<Commit>, error::Error> {
-    let project_id = project::ProjectId::from_str(id)?;
-    let project = project::Project::open(paths, &project_id)?;
-    let mut browser = match project {
-        project::Project::Git(git_project) => git_project.browser()?,
-    };
-    browser.branch(surf::git::BranchName::new(branch))?;
-
-    let commits = browser.get().iter().map(Commit::from).collect();
-
-    Ok(commits)
-}
-
-/// Retrieves the list of [`Tag`] for the given project `id`.
-///
-/// # Errors
-///
-/// Will return [`error::Error`] if the project doesn't exist or the surf interaction fails.
-pub fn tags(paths: &Paths, id: &str) -> Result<Vec<Tag>, error::Error> {
-    let project_id = project::ProjectId::from_str(id)?;
-    let project = project::Project::open(paths, &project_id)?;
-    let browser = match project {
-        project::Project::Git(git_project) => git_project.browser()?,
-    };
-
-    let tag_names = browser.list_tags()?;
-
-    let mut tags: Vec<Tag> = tag_names
-        .into_iter()
-        .map(|tag_name| Tag(tag_name.name().to_string()))
-        .collect();
-
-    tags.sort();
+    /// Get the project found at `project_urn`.
+    pub async fn get_project(
```

Comment:
Bump.

Reply:
So, I think this function will later be in `PeerApi` and we'll just call `api.get_project(project_urn.parse()?)` wherever it's needed. It's living in here now because of faking the resolver.